packages/wallet/api/src/marshal-contexts.js

```diff
@@ -0,0 +1,257 @@
+    };
+  };
+
+  return harden({
+    savePurseActions: makeSaver('purse', myData.purse),
+    savePaymentActions: makeSaver('payment', myData.payment),
+    /**
+     * @param {string} id
+     * @param {Purse} purse
+     */
+    initPurseId: (id, purse) => {
+      myData.purse.bySlot.init(id, purse);
+      myData.purse.byVal.init(purse, id);
+    },
+    purseEntries: myData.purse.bySlot.entries,
+    /**
+     * @param {string} id
+     * @param {unknown} val
+     */
+    initBoardId: (id, val) => {
+      sharedData.bySlot.init(id, val);
+      sharedData.byVal.init(val, id);
+    },
+    /**
+     * @param {string} id
+     * @param {unknown} val
+     */
+    ensureBoardId: (id, val) => {
+      if (sharedData.byVal.has(val)) {
+        assert.equal(sharedData.byVal.get(val), id);
+        return;
+      }
+      sharedData.bySlot.init(id, val);
+      sharedData.byVal.init(val, id);
+    },
+    ...makeMarshal(valToSlot, slotToVal),
+  });
+};
+
+const defaultMakePresence = iface => {
+  const severed = `SEVERED: ${iface.replace(/^Alleged: /, '')}`;
+  const thing = /** @type {any} */ (Far(severed, {}));
+  return thing;
+};
+
+/**
+ * Make context for unserializing wallet or board data.
+ *
+ * @param {(iface: string) => unknown} [makePresence]
+ */
+export const makeImportContext = (makePresence = defaultMakePresence) => {
+  // constraint: none of these keys has another as a prefix.
+  // and none overlaps with board
+  const myData = {
+    purse: {
+      /** @type {MapStore<string, PurseActions>} */
+      bySlot: makeScalarMap(),
+      /** @type {MapStore<PurseActions, string>} */
+      byVal: makeScalarMap(),
+    },
+    payment: {
+      /** @type {MapStore<string, PaymentActions>} */
+      bySlot: makeScalarMap(),
+      /** @type {MapStore<PaymentActions, string>} */
+      byVal: makeScalarMap(),
+    },
+    // TODO: 6 in total, right?
+  };
+  const sharedData = {
+    /** @type {MapStore<string, unknown>} */
+    bySlot: makeScalarMap(),
+    /** @type {MapStore<unknown, string>} */
+    byVal: makeScalarMap(),
+  };
+
+  const kindOf = slot => Object.keys(myData).find(k => slot.startsWith(k));
+
+  const slotToVal = {
+    /**
+     * @param {string} slot
+     * @param {string} iface
+     */
+    fromBoard: (slot, iface) => {
+      if (sharedData.bySlot.has(slot)) {
+        return sharedData.bySlot.get(slot);
+      }
+      const kind = kindOf(slot);
+      if (kind) {
+        assert.fail(X`bad shared slot ${q(slot)}`);
+      }
+      // TODO: assert(slot.startswith('board'))?
+      const it = makePresence(iface);
+
+      sharedData.bySlot.init(slot, it);
+      sharedData.byVal.init(it, slot);
+      return it;
+    },
+
+    /**
+     * @param {string} slot
+     * @param {string} iface
+     */
+    fromMyWallet: (slot, iface) => {
+      const kind = kindOf(slot);
+      const key = kind ? slot.slice(kind.length + 1) : slot;
+      const table = kind ? myData[kind] : sharedData;
```

Comment:
why does "MyWallet" include the board?

Reply:
so that we can recognize brands in the wallet as being identical to brands from the AMM (that's how we got started with this whole mess :)